src/javascript/typescript-config.ts

```diff
@@ -479,11 +595,89 @@ export class TypescriptConfig extends Component {
     }
   }
 
+  /**
+   * Resolve valid TypeScript extends paths relative to this config.
+   *
+   * @remarks
+   * This will only resolve the relative path from this config to another given
+   * an absolute path as input. Any non-absolute path or other string will be returned as is.
+   * This is to preserve manually specified relative paths as well as npm import paths.
+   *
+   * @param configPath Path to resolve against.
+   */
+  public resolveExtendsPath(configPath: string) {
+    // if not absolute assume user-defined path (or npm package).
+    if (!path.isAbsolute(configPath)) return configPath;
+    const relativeConfig = path.relative(
+      path.dirname(this.file.absolutePath),
+      configPath
+    );
+    // ensure immediate sibling files are prefixed with './'
+    // typescript figures this out, but some tools seemingly break without it (i.e, eslint).
+    const { dir, ...pathParts } = path.parse(relativeConfig);
+    const configDir = dir
+      ? path.format({ dir: dir.startsWith("..") ? "" : ".", base: dir })
+      : ".";
+    return path.format({ ...pathParts, dir: configDir });
+  }
+
+  /**
+   * Validate usage of `extends` against current TypeScript version.
+   * @private
+   */
+  private validateExtends() {
+    // accept no extends or singular extends.
+    if (this.extends.length <= 1) return;
+    const ts = this.project.deps.tryGetDependency("typescript");
+    const tsVersion = semver.coerce(ts?.version, { loose: true });
+    // we assume latest by default, so return if not set (or not valid).
+    if (!ts || !tsVersion) return;
+    if (tsVersion.major < 5) {
+      this.project.logger.warn(
+        "TypeScript >= 5.0.0 is required to extends from more than one base config.",
```

Comment:
The error sounds like it's the wrong way around:
```suggestion
        "TypeScript < 5.0.0 is can only extend from a single base config.",
```